gnovm/tests/stdlibs/std/std.go

```diff
@@ -209,26 +209,33 @@ func (tb *TestBanker) SendCoins(from, to crypto.Bech32Address, amt tm2std.Coins)
 	tb.CoinTable[from] = frest
 	// Second, add to 'to'.
 	// NOTE: even works when from==to, due to 2-step isolation.
-	tcoins, _ := tb.CoinTable[to]
+	tcoins := tb.CoinTable[to]
 	tsum := tcoins.Add(amt)
 	tb.CoinTable[to] = tsum
 }
 
 // TotalCoin implements the Banker interface.
 func (tb *TestBanker) TotalCoin(denom string) int64 {
-	panic("not yet implemented")
+	if denom == "" {
+		panic("empty denom")
+	}
+	var total int64
+	for _, coins := range tb.CoinTable {
+		total += coins.AmountOf(denom)
```

Comment:
still not good; we should not iterate through the whole table at each read, we should do the opposite and maintain a totalsupply value somewhere else; or just remove this feature.

Reply:
https://github.com/gnolang/gno/pull/3899/commits/6f00822048c99bdbff675fd94eae0e99415983ee

Instead of iterating, I added a `supplyStore` and modified it to directly retrieve the total supply of each denom from that store. Also, this will automatically update total supply when functions that affects the supply, such as `IssueCoin` and `RemoveCoin`, are called.

So, it only needs to retrieve values from a kv store, the operation of `TotalCoin` itself is reduce to constant time.